tools/lsp-wake/lsp.cpp

```diff
@@ -530,15 +530,29 @@ class LSPServer {
 
     void didChangeWatchedFiles(const JAST &receivedMessage) {
       JAST jfiles = receivedMessage.get("params").get("changes");
-      size_t changed = 0;
       for (auto child: jfiles.children) {
         std::string fileUri = child.second.get("uri").value;
-        changed += astree.changedFiles.erase(JSONConverter::decodePath(fileUri));
-      }
-      if (changed) {
-        needsUpdate = true;
-        refresh("watch-list-updated");
+        std::string filePath = JSONConverter::decodePath(fileUri);
+        astree.changedFiles.erase(filePath);
+
+        int eventType = stoi(child.second.get("type").value);
+        switch (eventType) {
+          case 1: // The file got created
+            needsUpdate = true;
+            break;
+          case 2: // The file got changed
```

Comment:
I think we need to update in this case, too. This indicates someone modified a file external to vscode. We need to refresh to see if that change affected our diagnosis.